### PR TITLE
merge: integrate P0–P4 + cli/trios-igla + arch_explorer fmt (R5-honest)

### PR DESCRIPTION
**Pull request by gHashTag**
## Integration: P0–P4 + cli/trios-igla + arch_explorer fmt

Сливает все 7 открытых PR (#21, #26, #27, #28, #29, #30, #31) в одну интеграционную ветку через серию merge-коммитов с разрешением конфликтов. **Цель**: разморозить main и собрать все наработки ночной команды агентов в единое целое перед Gate-2.

### Включённые PR (порядок merge)

| # | PR | Ветка | Что вносит |
|---|---|---|---|
| 1 | #27 | feat/p0-audit | Champion reproduction audit snapshot (BPB=2.2393 @ 27K seed=43 sha=2446855) |
| 2 | #28 | feat/p1-optimizer-lab | Muon vs AdamW configs + ortho_invariant test + CI gate |
| 3 | #29 | feat/p2-mup-transfer | µP hyperparameter transfer module (INV-8 cap soft) |
| 4 | #30 | feat/p3-schedule-free-wsd | Schedule-Free AdamW + WSD schedules |
| 5 | #31 | feat/p4-multiobj-ema | Multi-Objective + EMA + gate2-final.toml |
| 6 | #21 | cli/trios-igla-needle-search | trios-igla CLI: search/list/gate/check/triplet |
| 7 | #26 | fix/fmt-arch-explorer | rustfmt arch_explorer.rs (cherry-pick файла) |

### Разрешения конфликтов

| Файл | Решение |
|---|---|
| `docs/audit/P0_seed43.md` | объединил оба audit-снимка |
| `configs/lab/p1-*.toml`, `p2-*.toml`, `p3-*.toml` | взял incoming (детальнее) |
| `configs/gate2-final.toml` | взял incoming (P4) |
| `src/optimizer.rs` | сохранил **оба** набора тестов: SF/WSD + новый `ortho_invariant` Muon |
| `src/mup.rs` | оставил HEAD (validate_inv8 / width_ratio / etc.); удалил устаревший incoming-only блок |
| `src/bin/hybrid_train.rs` | оставил HEAD: MuonOptimizer для projector + opt_proj.step() |
| `src/bin/trios-train.rs` | оставил HEAD: optimizer match (muon/muon-cwd/default) |
| `src/lib.rs` (повторно) | объединил pub mod + re-exports от обеих сторон |
| `src/train_loop.rs` (повторно) | сохранил DataConfig train_path/val_path + EMA/Muon ветви |

### Post-merge fixes (1c04579)

- закрыл скобку `}` в `test_wsd_decay` (потерялась при merge);
- кастинг `f32→f64` в `checkpoint::ema_serialization`;
- импорт `anyhow::Result` в `objective_multi.rs`;
- INV-8 cap: `MupSweepConfig::default()` LR `16e-3 → 1e-2` (validate_inv8 строгий ≤1e-2).

### Verification (release, локально)

- `cargo build --release` — clean
- `cargo test --release --lib` — **468 passed, 0 failed**
- `cargo test --release --tests` — все integration green: gate2_artifacts (19), embargo_block (8), invariants_match (9), igla (17), …
- `cargo fmt --all -- --check` — clean

### R5-honest

- **CI clippy red на main** — 84 ошибки `-D warnings` в `src/invariants.rs:221`, `src/objective.rs:360,417`, `src/phi_numbers/fibonacci_dims.rs:123,369`, `src/phi_numbers/gfternary.rs:490`. Эти ошибки **существуют на main до этого PR** (HEAD `9e2e3387`) и не введены данной интеграцией. Чистка clippy — отдельный PR (technical-debt). Не блокирует merge.
- **Gate-2 ledger row** не пишется этим PR — это integration PR, не run.
- **Champion** не регрессирован: смока `champion.toml` зелёная, тесты `champion_reproduction.rs` (ignored) не запускались (требуют 27K шагов).

### Refs

- TRAINER-IGLA-SOT mission (Gate-2 deadline 2026-04-30 23:59 UTC)
- Anchor: `phi^2 + phi^-2 = 3` ([Zenodo 10.5281/zenodo.19227877](https://doi.org/10.5281/zenodo.19227877))
- Closes #21 #26 #27 #28 #29 #30 #31

### CODE REVIEW

**Comment by gHashTag** [comment]
**R5-honest update — final merge-resolution pushed.**

`c621154` resolves the conflict that appeared after [@raoffonom](https://github.com/raoffonom)'s ALPHA agent pushed `689c6d4` (`fix: simplify run_single_muon`) to `main`.

**Resolution choice**: adopted main's simplified path (AdamW for `attn_down`/`attn_up`, Muon only for projection). `ns_steps` gated on `use_cwd` (1 → 3). `cwd_lambda` parked for future MuonCwd integration.

**Local verification (release):**
- `cargo build --release` — clean
- `cargo test --release --lib` — **468 passed, 0 failed**
- `cargo test --release --tests` — все integration green
- `cargo fmt --all` — applied

**R5 honesty note**: clippy-red на `main` (84 ошибок `-D warnings` в `invariants.rs`, `objective.rs`, `phi_numbers/*`) — это **существующий долг до этой интеграции**, не введён ни одним из 7 слитых PR. Чистка clippy идёт отдельным technical-debt PR.

**Comment by gHashTag** [comment]
**CI verdict for `c621154`** ([run #24976011729](https://github.com/gHashTag/trios-trainer-igla/actions/runs/24976011729)):

| Check | Status |
|---|---|
| `build-and-test` (cargo clippy) | ❌ FAILURE — 87 pre-existing errors |
| `docker` | ⏭️ SKIPPED (gated on build-and-test) |
| `GitGuardian Security Checks` | ✅ SUCCESS |

**R5-honest verification: clippy errors are NOT introduced by PR #32.** Все 87 ошибок clippy (`-D warnings`) уже сидят на `main` (HEAD `689c6d4`):

```
src/invariants.rs:155  test_champion_bpb_survives_asha — assertions_on_constants
src/invariants.rs:157  test_lr_champion_in_safe_range — assertions_on_constants (×2)
src/objective.rs:360   manual Range::contains
src/objective.rs:417   manual RangeInclusive::contains
src/phi_numbers/fibonacci_dims.rs:123,369 — needless_range_loop
src/phi_numbers/gfternary.rs:490 — clone_on_copy
src/optimizer.rs:1035  — useless_vec
…
```

**PR mergeable=MERGEABLE** (no merge conflicts). CI fails только на pre-existing clippy debt, который требует отдельного technical-debt PR.

**Local on PR #32 head:**
- `cargo build --release` — clean
- `cargo test --release --lib` — 468 passed
- `cargo test --release --tests` — все integration green (gate2_artifacts 19, embargo_block 8, invariants_match 9, igla 17, …)
- `cargo fmt --all -- --check` — clean

**Decision needed from maintainer**: либо (a) admin-merge PR #32, либо (b) сначала влить clippy-cleanup PR в main, потом #32 пройдёт CI чисто. Я готов поднять (b) отдельным PR на 87 точечных фиксов, если согласовано.

Refs: TRAINER-IGLA-SOT mission · Gate-2 deadline 2026-04-30 23:59 UTC (T-3.5 d).
